File: smartbinapp/lib/database/mongodb_handler.dart
import 'package:mongo_dart/mongo_dart.dart';

class MongoDBHandler {
  final username = 'Shubass';
  final password = 'test1234';
  final String databaseName = 'smart_waste_management';
  final connectionString = 'mongodb://192.168.1.8:27017/';
  Db? db;

  Future<void> getDatabaseConnection() async {
    db = await Db.create("mongodb+srv://Shubass:test1234@cluster0.guugpkj.mongodb.net/smart_waste_management?retryWrites=true&w=majority");
    await db?.open();
  }
}

File: Desktop Application/SmartWasteManagement,Main/Database/MongoDBHandler.cs
using MongoDB.Driver;

namespace SmartWasteManagement.Main.Database
{
    public class MongoDBHandler
    {
        string ConnectionString { get; set; } = "mongodb+srv://Shubass:test1234@cluster0.guugpkj.mongodb.net/?retryWrites=true&w=majority";
        string UserName { get; set; } = "root";
        string Password { get; set; } = "root";

        const string DatabaseName = "smart_waste_management";

        public MongoClient MongoClient { get; set; }
        public IMongoDatabase Database { get; set; }

        public MongoDBHandler(string connectionString, string userName, string password)
        {
            ConnectionString = connectionString;
            UserName = userName;
            Password = password;
            MongoClient = new MongoClient(ConnectionString);
            Database = MongoClient.GetDatabase(DatabaseName);
        }

        public MongoDBHandler(string connectionString)
        {
            ConnectionString = connectionString;
            MongoClient = new MongoClient(ConnectionString);
            Database = MongoClient.GetDatabase(DatabaseName);
        }

        public MongoDBHandler()
        {
            MongoClient = new MongoClient(ConnectionString);
            Database = MongoClient.GetDatabase(DatabaseName);
        }

        public bool IsAlive()
        {
            throw new NotImplementedException();
        }
    }
}